packages/sns/lib/utils/snsInitter.ts

```diff
@@ -21,9 +30,9 @@ export async function initSnsSqs(
   extraParams?: ExtraParams,
 ) {
   if (!locatorConfig?.subscriptionArn) {
-    if (!creationConfig?.topic) {
+    if (!creationConfig?.topic && !locatorConfig?.topicArn && !locatorConfig?.topicName) {
       throw new Error(
-        'If locatorConfig.subscriptionArn is not specified, creationConfig.topic parameter is mandatory, as there will be an attempt to create the missing topic',
+        'If locatorConfig.subscriptionArn is not specified, creationConfig.topic, localtorConfig.name or locatorConfig.topicArn parameter is mandatory, as there will be an attempt to create the missing topic',
```

Comment:
Shouldn't we reword the message that attempt to create topic happens only if `creationConfig` is provided?

Reply:
yes

Reply:
Reworded.